http/auth_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"encoding/hex"
+	"testing"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/sourcenetwork/defradb/acp"
+	"github.com/sourcenetwork/defradb/crypto"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildAuthToken(t *testing.T) {
```

Comment:
question: How is authentication tested?  Is it just this file, or is it automagically covered by the http/cli integration tests?  Is authentication now always required when ACP is enabled?  Is there now no easy way for users of the embedded Go client to provide an ACP identity (is this untested?)?

Reply:
> How is authentication tested?

It's tested by the integration tests as well. Any place we are using an identity in a test will use authentication when using the http / cli clients.

> Is there now no easy way for users of the embedded Go client to provide an ACP identity (is this untested?)?

This doesn't change anything with how we use identity in the Go client.

```go
identity := acp.IdentityFromPublicKey(...)
ctx := db.SetContextIdentity(ctx, identity)
```